openedx/features/course_experience/tests/views/test_welcome_message.py

```diff
@@ -0,0 +1,90 @@
+from xmodule.modulestore.tests.factories import CourseFactory, ItemFactory
+from xmodule.html_module import CourseInfoModule
+
+TEST_PASSWORD = 'test'
+TEST_WELCOME_MESSAGE = '<h2>Welcome!</h2>';
+
+
+def welcome_message_url(course):
+    """
+    Returns the URL for the welcome message view.
+    """
+    return reverse(
+        'openedx.course_experience.welcome_message_fragment_view',
+        kwargs={
+            'course_id': unicode(course.id),
+        }
+    )
+
+
+def create_welcome_message(course, user, welcome_message=TEST_WELCOME_MESSAGE):
+    """
+    Creates a test welcome message for the specified course.
+    """
+    updates_usage_key = course.id.make_usage_key('course_info', 'updates')
+    course_updates = modulestore().create_item(
+        user.id,
+        updates_usage_key.course_key,
+        updates_usage_key.block_type,
+        block_id=updates_usage_key.block_id
+    )
+    course_updates.data = u'<ol><li><a href="test">Test Update</a></li></ol>'
+    course_updates.items.append({
+        "id": len(course_updates.items) + 1,
+        "date": 'December 31, 1999',
+        "content": welcome_message,
+        "status": CourseInfoModule.STATUS_VISIBLE
+    })
+    modulestore().update_item(course_updates, user.id)
+
+
+class TestWelcomeMessageView(SharedModuleStoreTestCase):
+    """
+    Tests for the course welcome message fragment view.
+    """
+    @classmethod
+    def setUpClass(cls):
+        """Set up the simplest course possible."""
+        # setUpClassAndTestData() already calls setUpClass on SharedModuleStoreTestCase
+        # pylint: disable=super-method-not-called
+        with super(TestWelcomeMessageView, cls).setUpClassAndTestData():
+            with cls.store.default_store(ModuleStoreEnum.Type.split):
+                cls.course = CourseFactory.create()
+                with cls.store.bulk_operations(cls.course.id):
+                    # Create a basic course structure
+                    chapter = ItemFactory.create(category='chapter', parent_location=cls.course.location)
+                    section = ItemFactory.create(category='sequential', parent_location=chapter.location)
+                    ItemFactory.create(category='vertical', parent_location=section.location)
+
+    @classmethod
+    def setUpTestData(cls):
+        """Set up and enroll our fake user in the course."""
+        cls.user = UserFactory(password=TEST_PASSWORD)
+        CourseEnrollment.enroll(cls.user, cls.course.id)
+
+    def setUp(self):
+        super(TestWelcomeMessageView, self).setUp()
+        self.client.login(username=self.user.username, password=TEST_PASSWORD)
+
+    def test_welcome_message(self):
+        create_welcome_message(self.course, self.user)
+        response = self.client.get(welcome_message_url(self.course))
+        self.assertEqual(response.status_code, 200)
+        self.assertContains(response, TEST_WELCOME_MESSAGE)
+
+    @skip('Disabled until web-fragments library supports status codes')
```

Comment:
What does this do now?  Return a 200?  Seems like it would be better to have a currently passing test for this scenario.

Then, I'd either have no comment, or a TODO comment with a ticket (that would probably be in the web-fragments library).